### PR TITLE
[experimental/python] Remove library mode from Python workflow

### PR DESCRIPTION
**Pull request by khalatepradnya**
Addressing one point in issue https://github.com/NVIDIA/cuda-quantum/issues/1251

### CODE REVIEW

File: python/cudaq/__init__.py [hunk 1]
```diff
@@ -8,16 +8,14 @@
 
 import sys, os
 from ._packages import *
-from .kernel.kernel_decorator import kernel, PyKernelDecorator, globalImportedKernels
+from .kernel.kernel_decorator import kernel, PyKernelDecorator
 from .kernel.kernel_builder import make_kernel, QuakeValue, PyKernel
 from .kernel.ast_bridge import globalAstRegistry, globalKernelRegistry
-from .kernel.qubit_qis import adjoint, control, compute_action
+from .kernel.qubit_qis import h, x, y, z, s, t, rx, ry, rz, r1, swap, exp_pauli, mx, my, mz, adjoint, control, compute_action
```
Comment (khalatepradnya) [inline python/cudaq/__init__.py:14]: This allows use of `cudaq.h`

Comment (khalatepradnya) [reply python/cudaq/__init__.py:14]: Pylance will be handled in a separate PR.

File: python/cudaq/kernel/kernel_decorator.py [hunk 2]
```diff
@@ -61,18 +45,9 @@ def __init__(self,
...
-            library_mode = False
-
-        # Library Mode
-        self.library_mode = library_mode
-        if jit == True:
-            self.library_mode = False
-
         if self.kernelFunction is None:
             if self.module is not None:
+                ## ASKME: Is the following still needed?
```
Comment (amccaskey) [inline python/cudaq/kernel/kernel_decorator.py:50]: Yes this is needed for `cudaq.synthesize()`

**Review by amccaskey — APPROVED** [review]
LGTM

**Pull request merged**